### PR TITLE
Add missing shelljs configuration options

### PR DESCRIPTION
**Pull request by voy**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/shelljs/shelljs (configuration section of the readme)
- [x] Increase the version number in the header if appropriate.
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

### CODE REVIEW

File: types/shelljs/index.d.ts [hunk 1]
```diff
@@ -537,10 +544,28 @@ export interface ShellConfig {
...
      * Absolute path of the Node binary. Default is null (inferred).
      * @type {string|null}
      */
     execPath: string | null;
+
+    /**
+     * Reset shell.config to the defaults.
+     * @type {ResetConfigFunction}
+     */
+    reset: ResetConfigFunction;
```
Comment (nikeee) [inline types/shelljs/index.d.ts]: Why did you create a new type instead of just defining this as `reset(): void;`?

**Review by nikeee — CHANGES_REQUESTED** [review]

File: types/shelljs/index.d.ts (hunk 1 above)
Comment (voy) [inline types/shelljs/index.d.ts]: my bad, I am gonna update the PR.

Comment (voy) [inline types/shelljs/index.d.ts]: Oh, in that case I am not sure what the type in the comment should be, though. I guess that's why I went for a new type.

Comment (nikeee) [inline types/shelljs/index.d.ts]: You could just use JSDoc to annotate the function parameters (none in this case) and the return type (`void`). No need for a `@type` in this case.
```
@returns {void}
```

**Review by nikeee — APPROVED** [review]

**Comment by nikeee** [comment]
:+1:

**Pull request merged**